org-code-javabuilder/lib/src/main/java/org/code/neighborhood/Direction.java

```diff
@@ -0,0 +1,63 @@
+package org.code.neighborhood;
+
+public enum Direction {
+  NORTH("north"),
+  EAST("east"),
+  SOUTH("south"),
+  WEST("west");
+  private String directionString;
+
+  Direction(String directionString) {
+    this.directionString = directionString;
+  }
+
+  public Direction turnLeft() {
+    Direction newDir;
+    if (this == NORTH) {
+      newDir = WEST;
+    } else {
+      newDir = this.values()[this.ordinal() - 1];
+    }
+    System.out.println("pointing " + newDir.getDirectionString());
+    return newDir;
+  }
+
+  public String getDirectionString() {
+    return directionString;
+  }
+
+  // Returns true if the current direction is north
+  public boolean isNorth() {
+    return this == NORTH;
+  }
+
+  // Returns true if the current direction is south
+  public boolean isSouth() {
+    return this == SOUTH;
+  }
+
+  // Returns true if the current direction is east
+  public boolean isEast() {
+    return this == EAST;
+  }
+
+  // Returns true if the current direction is west
+  public boolean isWest() {
+    return this == WEST;
+  }
+
+  public static Direction fromString(String text) {
+    if (text.equalsIgnoreCase("north")) {
+      return NORTH;
+    } else if (text.equalsIgnoreCase("east")) {
+      return EAST;
+    } else if (text.equalsIgnoreCase("south")) {
+      return SOUTH;
+    } else if (text.equalsIgnoreCase("west")) {
+      return WEST;
+    } else {
+      System.out.println("bad direction");
```

Comment:
Should this be an exception? (If yes, it doesn't necessarily need to be updated now - it can wait for the exception PR to be merged)

Reply:
It should be, I used exceptions everywhere else so I'll update this to match